compiler/src/iree/compiler/Preprocessing/Passes.cpp

```diff
@@ -134,10 +134,27 @@ buildTransposeConvolutionPassPipeline(OpPassManager &passManager,
 static void
 buildMakeSingleDispatchPassPipeline(OpPassManager &passManager,
                                     const TransformOptions &options) {
+  // We generalize certain named ops immediately before folding unit extent
+  // dims as the unit dim folding pass updates indexing maps and is better
+  // at working with generics. By this point we have already done any
+  // specialized raising and the op names are no longer useful.
+  passManager.addPass(GlobalOptimization::createGeneralizeLinalgNamedOpsPass());
+  passManager.addPass(DispatchCreation::createFoldUnitExtentDimsForFuncPass());
   GlobalOptimization::PropagateLinalgTransposePassOptions transposeOptions;
   transposeOptions.enableConvolutionPropagation = true;
+  transposeOptions.enableAggressivePropagation = true;
   passManager.addPass(
       GlobalOptimization::createPropagateLinalgTransposePass(transposeOptions));
+  passManager.addPass(GlobalOptimization::createGeneralizeLinalgNamedOpsPass());
```

Comment:
Is it necessary to duplicate this pass here?

Reply:
Yes transpose propagation can add back transposes and we need to generalize them again. The default path also does this in dispatch creation.